internal/proto/controller/api/resources/credentialstores/v1/credential_store.proto

```diff
@@ -7,81 +7,92 @@ option go_package = "github.com/hashicorp/boundary/sdk/pbs/controller/api/resour
 message CredentialStore {
   // Output only. The ID of the Credential Store.
-  string id = 10;
+  string id = 10; // @gotags: `class:"public"`
 
   // The ID of the Scope of which this Credential Store is a part.
-  string scope_id = 20 [json_name = "scope_id"];
+  string scope_id = 20 [json_name = "scope_id"]; // @gotags: `class:"public"`
 
   // Output only. Scope information for this Credential Store.
-  resources.scopes.v1.ScopeInfo scope = 30;
+  resources.scopes.v1.ScopeInfo scope = 30; // @gotags: `class:"public"`
 
   // Optional name for identification purposes.
-  google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "name" that: "Name" }];
+  google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "name" that: "Name" }]; // @gotags: `class:"public"`
 
   // Optional user-set description for identification purposes.
-  google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "description" that: "Description" }];
+  google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "description" that: "Description" }]; // @gotags: `class:"public"`
 
   // Output only. The time this resource was created.
-  google.protobuf.Timestamp created_time = 60 [json_name = "created_time"];
+  google.protobuf.Timestamp created_time = 60 [json_name = "created_time"]; // @gotags: `class:"public"`
 
   // Output only. The time this resource was last updated.
-  google.protobuf.Timestamp updated_time = 70 [json_name = "updated_time"];
+  google.protobuf.Timestamp updated_time = 70 [json_name = "updated_time"]; // @gotags: `class:"public"`
 
   // Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.
   // The mutation will fail if the version does not match the latest known good version.
-  uint32 version = 80;
+  uint32 version = 80; // @gotags: `class:"public"`
 
   // The Credential Store type.
-  string type = 90;
-
-  // The attributes that are applicable for the specific Credential Store type.
-  google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true];
+  string type = 90; // @gotags: `class:"public"`
+
+  oneof attrs {
+    // The attributes that are applicable for the specific Credential Store type.
+    google.protobuf.Struct attributes = 100 [
+      (custom_options.v1.generate_sdk_option) = true,
+      (custom_options.v1.subtype) = "default"
+    ]; // @gotags: `class:"public"`
+    VaultCredentialStoreAttributes vault_credential_store_attributes = 101 [
+      (google.api.field_visibility).restriction = "INTERNAL",
+      (custom_options.v1.generate_sdk_option) = true,
+      (custom_options.v1.subtype) = "vault"
+    ]; // @gotags: `class:"public"`
```

Comment:
It's a bit confusing to see `public` here while some values inside the struct are not `public`. What does `public` convey here -- the default status for anything not explicitly overridden inside the struct?

Reply:
Talked to @jimlambrt  -- tagging the structure here will have no effect whatsoever. Might be worth removing tags on the structures in favor of just the items inside.

Reply:
Will do